CAS doesn't work with Spring 4

CAS compiles but won't start with Spring 4.
Need to apply #2491 (fix for #2461) before, otherwise the spring config error will show up before.

The `org.jasig.cas:*:4.0.0` dependencies we're using seem to be just too old ([May 2014](https://mvnrepository.com/artifact/org.jasig.cas/cas-server-core/4.0.0)) and moved from the `org.jasig.cas` namespace to [org.apereo.cas](https://mvnrepository.com/artifact/org.apereo.cas/cas-server-core).

I'm afraid this might not be just a trivial dependency upgrade though and blocks 18.12. Sigh.

@fvanderbiest @pmauduit @severo shall we panic? I think I'm hyperventilating already. It doesn't help that the cas module has no tests at all, guess we should at the very least add some smoke tests to make sure the app starts up and listens.

ah, forgot to mention which errors it's throwing:

```
Caused by: org.springframework.beans.BeanInstantiationException: Failed to instantiate [org.springframework.scheduling.quartz.MethodInvokingJobDetailFactoryBean]: No default constructor found; nested exception is java.lang.NoClassDefFoundError: org/quartz/impl/JobDetailImpl
    at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate (SimpleInstantiationStrategy.java:85)
....
    at org.jasig.cas.web.init.SafeContextLoaderListener.contextInitialized (SafeContextLoaderListener.java:75)
....
Caused by: java.lang.NoClassDefFoundError: org/quartz/impl/JobDetailImpl
    at java.lang.Class.getDeclaredConstructors0 (Native Method)
    at java.lang.Class.privateGetDeclaredConstructors (Class.java:2671)
    at java.lang.Class.getConstructor0 (Class.java:3075)
    at java.lang.Class.getDeclaredConstructor (Class.java:2178)
    at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate (SimpleInstantiationStrategy.java:80)
....
    at org.jasig.cas.web.init.SafeContextLoaderListener.contextInitialized (SafeContextLoaderListener.java:75)

```

--- a/cas-server-webapp/pom.xml
+++ b/cas-server-webapp/pom.xml
@@ -30,6 +30,25 @@
   <properties>
     <packageDatadirScmVersion>18.12</packageDatadirScmVersion>
     <server>generic</server>
+    
+    <!-- spring version revert start -->
+    <!-- 
+    	See https://github.com/georchestra/georchestra/issues/2492
+    	cas-server-webapp doesn't work with Spring 4. 
+    	An upgrade of org.jasig.cas:*:4.0.0 is required.
+    	is required. https://mvnrepository.com/artifact/org.jasig.cas/cas-server-core/4.2.7 is a possibility,
+    	although latest version is from 2016.
+    	The project moved from org.jasig.cas to org.apereo.cas since then.
+    	Upgrading to https://mvnrepository.com/artifact/org.apereo.cas/cas-server-core/5.3.8
+    	would be preferred as it works with Spring 4.x, or 
+    	https://mvnrepository.com/artifact/org.apereo.cas/cas-server-core/6.0.1
+    	for Spring 5.x
+    -->
+    <spring.version>3.2.6.RELEASE</spring.version>
+    <spring-security.version>3.2.0.RELEASE</spring-security.version>
+    <hibernate.version>4.1.0.Final</hibernate.version>
+    <hibernate-validator.version>4.2.0.Final</hibernate-validator.version>
+    <!-- spring version revert end -->
   </properties>
   <dependencies>
     <dependency>

